addons/stock/models/stock_warehouse.py

```diff
@@ -1070,4 +1070,4 @@ def action_view_all_routes(self):
         }
 
     def get_current_warehouses(self):
-        return self.env['stock.warehouse'].search_read(fields=['id', 'name', 'code'], order='name')
+        return self.env['stock.warehouse'].search_read(fields=['id', 'name', 'code'], order='sequence')
```

Comment:
Leave `order` empty to use the model order (which is `sequence,id` so deterministic)